esp/esp/program/modules/handlers/jsondatamodule.py

```diff
@@ -689,6 +689,16 @@ def message_requests():
 
     message_requests.method.cached_function.depend_on_model('dbmail.MessageRequest')
 
+    @staticmethod
+    def calc_hours(classes):
+        hours = {"class-hours": 0, "class-student-hours": 0, "class-registered-hours": 0}
+        for cls in classes:
+            if cls['subject_duration']:
+                hours["class-hours"] += float(cls['subject_duration'])
+                hours["class-student-hours"] += float(cls['subject_duration']) * (float(cls['class_size_max']) if cls['class_size_max'] else 0)
+                hours["class-registered-hours"] += float(cls['subject_duration']) * float(cls['subject_students']) / float(cls['num_sections'])
```

Comment:
Why do we divide by the number of sections here?

Reply:
That is a good question...

Reply:
Oh, I just remembered why: `subject_duration` is the sum of the durations of the sections of the class. Each student only attends 1 section, so we need to divide `subject_duration` by `num_sections` to get the actual duration of the class that the student is registered for.